ksql-parser/src/main/java/io/confluent/ksql/parser/properties/with/WithClauseProperties.java

```diff
@@ -86,6 +88,32 @@ void validateDateTimeFormat(final String configName) {
     }
   }
 
+  static void validateWindowSizeProperty(final String windowSizeProperty) {
+    Objects.requireNonNull(windowSizeProperty, "windowSizeProperty");
+    final String[] sizeParts = windowSizeProperty.split(" ");
+    if (sizeParts.length != 2) {
+      throwWindowSizeException(windowSizeProperty);
+    }
+    try {
+      Long.parseLong(sizeParts[0]);
+    } catch (final NumberFormatException nfe) {
+      throwWindowSizeException(windowSizeProperty);
+    }
+
+    try {
+      TimeUnit.valueOf(sizeParts[1].toUpperCase());
```

Comment:
We should extend this to handle cases like  `1 SECOND`, i.e. singular `SECOND` rather than `SECONDS`.